azure-pipelines/main-pipeline.yml

```diff
@@ -0,0 +1,228 @@
+                      Write-Output '##vso[task.setvariable variable=accounts;isOutput=true]True'
+                      Write-Output '##vso[task.setvariable variable=commerce;isOutput=true]True'
+                      Write-Output '##vso[task.setvariable variable=inventory;isOutput=true]True'
+                  }
+                  'commerce/*' { Write-Output '##vso[task.setvariable variable=commerce;isOutput=true]True' }
+                  'config/*' { Write-Output '##vso[task.setvariable variable=config;isOutput=true]True' }
+                  'eureka-server/*' { Write-Output '##vso[task.setvariable variable=eureka-server;isOutput=true]True' }
+                  'inventory/*' { Write-Output '##vso[task.setvariable variable=inventory;isOutput=true]True' }
+                  'Spring-Cloud-Gateway/*' { Write-Output '##vso[task.setvariable variable=Spring-Cloud-Gateway;isOutput=true]True' }
+              }
+          }
+      name: ProduceVar
+
+# Printing a To-Do (Build) list. Note, that these are dynamic variables and if no variables exist
+# The script will print literal.
+
+    - script: |
+          echo 'Base will rebuild: $(ProduceVar.base)'
+          echo 'Accounts will rebuild: $(ProduceVar.accounts)'
+          echo 'Commerce will rebuild: $(ProduceVar.commerce)'
+          echo 'Inventory will rebuild: $(ProduceVar.inventory)'
+          echo 'Config will rebuild: $(ProduceVar.config)'
+          echo 'Eureka Server will rebuild: $(ProduceVar.eureka-server)'
+          echo 'Spring Cloud Gateway will rebuild: $(ProduceVar.Spring-Cloud-Gateway)'
+
+# This job ensures Base is built first, in case of dependencies.
+# Setting a variable with the output variable from the previous job.
+# It then checks that it's true, and continues on with the build, or skips.
+
+  - job: BaseBuild
+    pool:
+      name: java-16
+    dependsOn: CheckChanges
+    variables:
+        base: $[ dependencies.CheckChanges.outputs['ProduceVar.base'] ]
+    condition: eq(variables['base'], 'True')
+    steps:
+      - script: echo 'Base has been updated - $(base)'
+        displayName:  'Checking Base'
+
+# Sonar Cloud Prep 
+
+      - task: SonarCloudPrepare@1
+        inputs:
+          SonarCloud: 'SonarCloud'
+          organization: '$(org)'
+          scannerMode: 'CLI'
+          configMode: 'manual'
+          cliProjectKey: '$(projectKey)'
+          cliProjectName: '$(projectName)'
+          cliSources: '.'
+          extraProperties: 'sonar.java.binaries=**/target/classes'
+
+# Install Base service.
+
+      - script: 'mvn clean package -f ./base/pom.xml '
+
+# Sonar Analysis
+
+      - task: SonarCloudAnalyze@1
+      - task: SonarCloudPublish@1
+        inputs:
+          pollingTimeoutSec: '300'
+          
+      - task: PublishPipelineArtifact@1
+        inputs:
+          targetPath: '$(Build.Repository.LocalPath)/base/target/base-1.jar'
+          artifact: 'base'
+          publishLocation: 'pipeline'
+
+      - script:  echo 'Built base'
```

Comment:
steps like this are trying to do something which is better accomplished by just giving all steps, jobs, stages, etc good display names and letting the GUI display them accordingly

Reply:
Noted.